### PR TITLE
[Dev 224192] proof of concept sdk for all lightbox secure in app browsers

### PR DESCRIPTION
**Pull request by marcosrivereto**
** DO NOT MERGE THIS PR ** 

This PR is only to code validation

### CODE REVIEW

File: Example/TrustlySDK/ViewController.swift [hunk 1]
```diff
@@ -67,9 +68,24 @@ class ViewController: BaseViewController {
...
+
+        }
+
+        self.openLightbox()
         
-        let trustlyLightboxViewController = TrustlyLightBoxViewController()
-        trustlyLightboxViewController.delegate = self
+    }
+    
+    @IBAction func payWituAESWebAuth(_ sender: Any) {
```
Comment (renatodelpupo) [inline Example/TrustlySDK/ViewController.swift]: Maybe there's a typo here in the word `with`?

File: Sources/TrustlySDK/Utils/EstablishDataUtils.swift [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+            }
+            
+        }
+        
+        return newEstablish
+        
+    }
+    
+    /// Responsible to parse the dot notation key, in a clena dictionary key. Example: from ["customer.address.country": "US"] to ["customer":  ["address": [country": "US"]]]
```
Comment (renatodelpupo) [inline Sources/TrustlySDK/Utils/EstablishDataUtils.swift]: Maybe there's a typo here in the word `clean`?

File: Sources/TrustlySDK/Utils/EstablishDataUtils.swift [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    ///
+    /// - Parameters:
+    ///     - origin: The origin dicitionary
```
Comment (renatodelpupo) [inline Sources/TrustlySDK/Utils/EstablishDataUtils.swift]: Maybe there's a typo here in the word `dictionary`?

File: Sources/TrustlySDK/Utils/EstablishDataUtils.swift [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    ///     - key: The key with dot notation
+    ///     - value: The value of the key
+    static private func parseData(origin: inout [String : AnyHashable], key: String, value: AnyHashable ) {
+        
+        if key.contains(".") {
+            origin.removeValue(forKey: key)
+            
+            let splitedKeys = splitDotNotationKeys(key)
```
Comment (renatodelpupo) [inline Sources/TrustlySDK/Utils/EstablishDataUtils.swift]: Maybe there's a typo here in the word `splitted`? I even suggest `splitKeys` instead.

File: Sources/TrustlySDK/Utils/EstablishDataUtils.swift [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            parseData(origin: &newObject, key: splitedKeys.complementKey, value: value)
+            origin[splitedKeys.mainKey] = newObject
+            
+        } else {
+            origin[key] = value
+            
+        }
+    }
+    
+    /// Split the dot notation key oin 2 parts, mainKey and complementKey. Example: from "customer.address.country" to (mainKey: customer, complementKey: address.country)
```
Comment (renatodelpupo) [inline Sources/TrustlySDK/Utils/EstablishDataUtils.swift]: Maybe there's a typo here in the word `in`?

File: Sources/TrustlySDK/Utils/JSONUtils.swift [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+//  TrustlySDK
+//
+//  Created by Marcos Rivereto on 08/02/24.
+//
+
+import Foundation
+
+struct JSONUtils {
+    
+    static private func getJsonDataFrom(dicticionary: [String : AnyHashable]) -> Data? {
```
Comment (renatodelpupo) [inline Sources/TrustlySDK/Utils/JSONUtils.swift]: Maybe there's a typo here in the word `dictionary`?

File: Example/TrustlySDK/Base.lproj/Main.storyboard [hunk 7]
```diff
@@ -45,19 +45,18 @@
...
                                 </constraints>
                                 <fontDescription key="fontDescription" type="boldSystem" pointSize="17"/>
                                 <inset key="imageEdgeInsets" minX="0.0" minY="0.0" maxX="2.2250738585072014e-308" maxY="0.0"/>
-                                <state key="normal" title="Pay">
+                                <state key="normal" title="Pay with AESWebAuth">
                                     <color key="titleColor" systemColor="secondarySystemBackgroundColor"/>
                                 </state>
                                 <connections>
-                                    <action selector="pay:" destination="vXZ-lx-hvc" eventType="touchUpInside" id="eGD-FN-iNW"/>
+                                    <action selector="payWituAESWebAuth:" destination="vXZ-lx-hvc" eventType="touchUpInside" id="LyG-e4-6SZ"/>
```
Comment (renatodelpupo) [inline Example/TrustlySDK/Base.lproj/Main.storyboard]: Maybe there's a typo here in the word `with`?

**Review by renatodelpupo — CHANGES_REQUESTED** [review]

File: Example/TrustlySDK/ViewController.swift (hunk 1 above)
Comment (marcosrivereto) [reply Example/TrustlySDK/ViewController.swift]: Fixed

File: Sources/TrustlySDK/Utils/EstablishDataUtils.swift (hunk 3 above)
Comment (marcosrivereto) [reply Sources/TrustlySDK/Utils/EstablishDataUtils.swift]: Fixed

File: Sources/TrustlySDK/Utils/EstablishDataUtils.swift (hunk 5 above)
Comment (marcosrivereto) [reply Sources/TrustlySDK/Utils/EstablishDataUtils.swift]: Fixed

File: Sources/TrustlySDK/Utils/EstablishDataUtils.swift (hunk 2 above)
Comment (marcosrivereto) [reply Sources/TrustlySDK/Utils/EstablishDataUtils.swift]: Fixed

File: Sources/TrustlySDK/Utils/EstablishDataUtils.swift (hunk 4 above)
Comment (marcosrivereto) [reply Sources/TrustlySDK/Utils/EstablishDataUtils.swift]: Fixed

File: Sources/TrustlySDK/Utils/JSONUtils.swift (hunk 6 above)
Comment (marcosrivereto) [reply Sources/TrustlySDK/Utils/JSONUtils.swift]: Fixed

File: Example/TrustlySDK/Base.lproj/Main.storyboard (hunk 7 above)
Comment (marcosrivereto) [reply Example/TrustlySDK/Base.lproj/Main.storyboard]: Fixed

**Review by andresguedes — APPROVED** [review]

**Review by renatodelpupo — APPROVED** [review]

**Review by Aga92-92 — APPROVED** [review]

**Review by Aga92-92 — APPROVED** [review]

**Pull request closed without merging**